rpc/rpc.go

```diff
@@ -1092,22 +1092,25 @@ func (c *Conn) handleFinish(ctx context.Context, id answerID, releaseResultCaps
 		ans.cancel()
 	}
 	if ans.flags&returnSent == 0 {
-		c.mu.Unlock()
-		return nil
+		return
 	}
 
 	// Return sent and finish received: time to destroy answer.
-	rl, err := ans.destroy()
-	c.mu.Unlock()
-	if ans.releaseMsg != nil {
-		ans.releaseMsg()
+	delete(c.answers, ans.id)
+
+	if ans.flags&releaseResultCapsFlag != 0 {
+		for id, count := range ans.exportRefs {
+			if _, e := c.releaseExport(id, count); err == nil && e != nil {
+				err = rpcerr.Annotate(err, "incoming finish: release result caps")
+			}
+		}
```

Comment:
Looks like you inlined the body of ans.destroy() and then tweaked?

...I feel like it's worth auditing the other calls to .destroy() to make sure they're not doing the same wrong thing with the returned releaseList -- and maybe just change destory() and keep using it here?